central/complianceoperator/v2/rules/datastore/datastore_impl_test.go

```diff
@@ -1 +1,368 @@
+			expectedRecordIndex: set.NewFrozenIntSet(0, 1, 2),
+		},
+		{
+			desc: "Write 3 clusters - No access",
+			rules: []*storage.ComplianceOperatorRuleV2{
+				getTestRule(testconsts.Cluster1),
+				getTestRule(testconsts.Cluster2),
+				getTestRule(testconsts.Cluster3),
+			},
+			testContext:         s.noAccessCtx,
+			expectedRecordIndex: set.NewFrozenIntSet(),
+		},
+		{
+			desc: "Write 3 clusters - Cluster 1 access",
+			rules: []*storage.ComplianceOperatorRuleV2{
+				getTestRule(testconsts.Cluster1),
+				getTestRule(testconsts.Cluster2),
+				getTestRule(testconsts.Cluster3),
+			},
+			testContext:         s.testContexts[testutils.Cluster1ReadWriteCtx],
+			expectedRecordIndex: set.NewFrozenIntSet(0),
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Require().NoError(s.dataStore.UpsertRules(s.hasWriteCtx, tc.rules))
+
+		for _, rule := range tc.rules {
+			s.Require().NoError(s.dataStore.DeleteRule(tc.testContext, rule.GetId()))
+		}
+
+		count, err := s.storage.Count(s.hasReadCtx)
+		s.Require().NoError(err)
+		// If we could not delete the rules then they will remain.
+		s.Require().Equal(len(tc.rules)-tc.expectedRecordIndex.Cardinality(), count)
+
+		// Clean up
+		for _, rule := range tc.rules {
+			s.Require().NoError(s.dataStore.DeleteRule(s.hasWriteCtx, rule.GetId()))
+		}
+	}
+}
+
+func (s *complianceRuleDataStoreTestSuite) TestGetRulesByCluster() {
+	// make sure we have nothing
+	ruleIDs, err := s.storage.GetIDs(s.hasReadCtx)
+	s.Require().NoError(err)
+	s.Require().Empty(ruleIDs)
+
+	testRule1 := getTestRule(testconsts.Cluster1)
+	testRule2 := getTestRule(testconsts.Cluster1)
+	testRule3 := getTestRule(testconsts.Cluster2)
+
+	s.Require().NoError(s.dataStore.UpsertRule(s.hasWriteCtx, testRule1))
+	s.Require().NoError(s.dataStore.UpsertRule(s.hasWriteCtx, testRule2))
+	s.Require().NoError(s.dataStore.UpsertRule(s.hasWriteCtx, testRule3))
+
+	count, err := s.storage.Count(s.hasReadCtx)
+	s.Require().NoError(err)
+	s.Require().Equal(3, count)
+
+	testCases := []struct {
+		desc            string
+		clusterID       string
+		testContext     context.Context
+		expectedResults []*storage.ComplianceOperatorRuleV2
+		expectedCount   int
+	}{
+		{
+			desc:            "Rules exist - Full access",
+			clusterID:       testconsts.Cluster1,
+			testContext:     s.testContexts[testutils.UnrestrictedReadCtx],
+			expectedResults: []*storage.ComplianceOperatorRuleV2{testRule1, testRule2},
+			expectedCount:   2,
+		},
+		{
+			desc:            "Rules exist - Cluster 1 access",
+			clusterID:       testconsts.Cluster1,
+			testContext:     s.testContexts[testutils.Cluster1ReadWriteCtx],
+			expectedResults: []*storage.ComplianceOperatorRuleV2{testRule1, testRule2},
+			expectedCount:   2,
+		},
+		{
+			desc:            "Rules exist - Cluster 2 access",
+			clusterID:       testconsts.Cluster1,
+			testContext:     s.testContexts[testutils.Cluster2ReadWriteCtx],
+			expectedResults: nil,
+			expectedCount:   0,
```

Comment:
Just to make sure I understand why this is 0, is because we're asking for the rules for `Cluster1` with the wrong permissions since we're using the context for `Cluster2`, thus we get back 0 instead of 1, right?

Reply:
Correct.